corehq/apps/domain/views/settings.py

```diff
@@ -377,21 +377,21 @@ def post(self, request, *args, **kwargs):
     def page_context(self):
         apps = get_apps_in_domain(self.domain, include_remote=False)
         case_types = {t for app in apps for t in app.get_case_types() if t}
-        current_values = CaseSearchConfig.objects.get_or_none(pk=self.domain)
+        config = CaseSearchConfig.objects.get_or_none(pk=self.domain) or CaseSearchConfig(domain=self.domain)
```

Comment:
Yep, exactly. While case search is feature flagged, anyone visiting this page is likely going to create a config, but in the glorious future when case search is GA, it'd be nice if we didn't create a new object just because someone stumbled onto this page.